src/main/java/uk/gov/hmcts/cmc/claimstore/services/OffersService.java

```diff
@@ -119,6 +123,14 @@ private String userAction(String userAction, String userType) {
     public Claim signSettlementAgreement(String externalId, Settlement settlement, String authorisation) {
         final Claim claim = claimService.getClaimByExternalId(externalId, authorisation);
         assertSettlementIsNotReached(claim);
+
+        settlement.getLastOfferStatement().getOffer().ifPresent(offer ->
+            offer.getPaymentIntention().ifPresent(
+                paymentIntention ->
+                    claimantRepaymentPlanRule.assertClaimantRepaymentPlanIsValid(claim,
+                        paymentIntention.getRepaymentPlan().orElse(null)))
+        );
+
```

Comment:
As mentioned above, maybe we should move this logic - we'll have a chat whenever you have a chance.